mlrun/db/httpdb.py

```diff
@@ -3062,6 +3062,145 @@ def set_schedule_notifications(
+
+    def get_workflow_id(
+        self,
+        project: str,
+        name: str,
+        run_id: str,
+        engine: str = "",
+    ):
+        """
+        Retrieve workflow id from the uid of the workflow runner.
+
+        :param project: project name
+        :param name:    workflow name
+        :param run_id:  the id of the workflow runner - the job that runs the workflow
+        :param engine:  pipeline runner
+
+        :returns:   :py:class:`~mlrun.common.schemas.GetWorkflowResponse`.
+        """
+        params = {}
+        if engine:
+            params["engine"] = engine
+        response = self.api_call(
+            "GET",
+            f"projects/{project}/workflows/{name}/references/{run_id}",
+            params=params,
+        )
+        return mlrun.common.schemas.GetWorkflowResponse(**response.json())
+
+    def load_project(
+        self,
+        name: str,
+        url: str,
+        secrets: Optional[Dict] = None,
+        save_secrets: bool = True,
+    ) -> mlrun.common.schemas.BackgroundTask:
+        """
+        Loading a project remotely from the given source.
+        :param name:            project name
+        :param url:             git or tar.gz or .zip sources archive path e.g.:
+                                git://github.com/mlrun/demo-xgb-project.git
+                                http://mysite/archived-project.zip
+                                The git project should include the project yaml file.
+        :param secrets:         Secrets to store in project in order to load it from the provided url.
+                                For more information see :py:func:`mlrun.load_project` function.
+        :param save_secrets:    Whether to store secrets in the loaded project.
+                                Setting to False will cause waiting for the process completion.
+
+        :returns:      A BackgroundTask object, with details on execution process and its status.
+        """
+        params = {"url": url}
+        body = None
+        if secrets:
+            provider = mlrun.common.schemas.SecretProviderName.kubernetes
+            secrets_input = mlrun.common.schemas.SecretsData(
+                provider=provider, secrets=secrets
+            )
+            body = secrets_input.dict()
+        response = self.api_call(
+            "POST", f"projects/{name}/load", params=params, body=dict_to_json(body)
+        )
+
+        if secrets and not save_secrets:
+            # In order to remove secrets from project we need to wait for the background task to end:
+            bg_task = mlrun.common.schemas.BackgroundTask(**response.json())
+            while (
+                bg_task.status.state
+                not in mlrun.common.schemas.BackgroundTaskState.terminal_states()
+            ):
+                time.sleep(2)
+                bg_task = self.get_project_background_task(
+                    project=name, name=bg_task.metadata.name
+                )
```

Comment:
Use `retry_until_successful` and/or use a timeout here (I know there's an internal timeout for the load_project function but I think to cover all grounds it's best to use it plus some extra padding here)